simulated_beacon/api.go

```diff
@@ -0,0 +1,24 @@
+package simulated_beacon
+
+import (
+	"context"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+)
+
+type API struct {
+	simBeacon *SimulatedBeacon
+}
+
+func (api *API) AddWithdrawal(ctx context.Context, withdrawal *types.Withdrawal) error {
+	api.simBeacon.mu.Lock()
+	defer api.simBeacon.mu.Unlock()
+	return api.simBeacon.withdrawals.add(withdrawal)
+}
```

Comment:
This is IMO a bit odd. I guess you wanted to split this up into a standalone API in order to make the distinction more clear about what is the public API-surface and what is the actual implementation? 

The thing which is a bit odd, IMO, is that you have this external wrapper which reaches in and locks/unlocks an internal mutex inside teh `simBeacon`. Either move the mutex to this `API`, or, probably better, move the locking so the `SimulatedBeacon` controls it's own locks.

Reply:
So this type would be a 'pure' wrapper: 
```suggestion
func (api *API) AddWithdrawal(ctx context.Context, withdrawal *types.Withdrawal) error {
	return api.simBeacon.withdrawals.add(withdrawal)
}
```
You could even (maybe?) turn this whole file into 
```golang
type API SimulatedBeacon

func (api *API) AddWithdrawal(ctx context.Context, withdrawal *types.Withdrawal) error {
	return api.withdrawals.add(withdrawal)
}

func (api *API) SetFeeRecipient(ctx context.Context, feeRecipient *common.Address) {
	api.feeRecipient = *feeRecipient
}
```
Depends on your preference I guess, but it's less code at least

The last change also requires a change here: 
```diff 
--- a/simulated_beacon/simulated_beacon.go
+++ b/simulated_beacon/simulated_beacon.go
@@ -216,7 +216,7 @@ func RegisterAPIs(stack *node.Node, c *SimulatedBeacon) {
        stack.RegisterAPIs([]rpc.API{
                {
                        Namespace: "dev",
-                       Service:   &API{c},
+                       Service:   (*API)(c),
                        Version:   "1.0",
                },
        })
```

Reply:
I've moved the mutex-handling for controlling access for fee recipient and the withdrawals list into `SimulatedBeacon`/`withdrawals` and pulled them out of the api.  This should make things a bit cleaner.